types/react-relay/test/relay-experimental-tests.tsx

```diff
@@ -957,27 +974,241 @@ function LoadQuery() {
+        };
+    }
+
+    interface SomeQueryVariables {
+        slug: string;
+    }
+
+    interface SomeQuery {
+        readonly response: SomeQueryResponse;
+        readonly variables: SomeQueryVariables;
+    }
+
+    const query = graphql`
+        query SomeQuery($slug: String!) {
+            post {
+                name
+            }
+        }
+    `;
+
+    // RootEntrypoint.entrypoint
+    const RootEntrypointComponent: EntryPointComponent<{ someQueryRef: SomeQuery }, {}, { onClick: () => void }> = ({
+        queries,
+        props,
+    }) => {
+        const data = usePreloadedQuery(query, queries.someQueryRef);
+
+        return <button onClick={props.onClick}>{data.post.name}</button>;
+    };
+
+    interface Params {
+        slug: string;
+    }
+
+    const entrypoint: EntryPoint<typeof RootEntrypointComponent, Params> = {
+        root: JSResource<typeof RootEntrypointComponent>(),
+        getPreloadProps(
+            // $ExpectType Params
+            params,
+        ) {
+            return {
+                queries: {
+                    someQueryRef: {
+                        parameters: query as any,
+                        variables: {
+                            slug: params.slug,
+                        },
+                    },
+                },
+            };
+        },
+    };
+
+    const entrypointReference = loadEntryPoint(environmentProvider, entrypoint, {
+        slug: 'test-slug',
+    });
+
+    // $ExpectType SomeQueryVariables
+    const vars = entrypointReference.queries.someQueryRef.variables;
+
+    /*
+    EntrypointContainer Tests
+     */
+    function EntryPointContainerTests() {
         return (
-            <ShouldPassIntoUsePreloadQuery preloadedQuery={preloadedQuery} />
+            <EntryPointContainer
+                entryPointReference={entrypointReference}
+                props={{
+                    onClick() {},
+                }}
+            />
+        );
+    }
+
+    /*
+    EntryPointContainerNested Tests
+     */
+    function EntryPointContainerNested() {
+        const entrypointA: EntryPoint<typeof RootEntrypointComponent, Params> = {
+            root: JSResource(),
+            getPreloadProps(params) {
+                return {
+                    queries: {
+                        someQueryRef: {
+                            parameters: query as any,
+                            variables: {
+                                slug: params.slug,
+                            },
+                        },
+                    },
+                };
+            },
+        };
+
+        const entrypointB: EntryPoint<typeof RootEntrypointComponent, { author: string }> = {
+            root: JSResource(),
+            getPreloadProps(params) {
+                console.log(params.author);
+                return {
+                    queries: {
+                        someQueryRef: {
+                            parameters: query as any,
```

Comment:
Ditto.

Reply:
Currently in OSS relay-compiler we don't have the types or generated files for EntryPoint params. But its a `$parameters` file. So im not sure what to type it as. But the current `ConcreteRequest` works.

Maybe since i do accept that as a type, this `as any` isnt required. Ill remove, and make sure tests pass.